nexus/src/app/instance_platform/cpu_platform.rs

```diff
@@ -0,0 +1,1193 @@
+
+    // Hide extended cache topology as well (Leaf 8000_001D)
+    cpuid.set_extended_cache_parameters(None).expect("can set leaf 8000_001Dh");
+
+    let mut leaf = ProcessorTopologyInfo::empty();
+    // This is managed dynamically, where a single vCPU instance will see this
+    // set to 1 instead.
+    leaf.set_threads_per_core(2);
+    cpuid
+        .set_processor_topology_info(Some(leaf))
+        .expect("can set leaf 8000_001Eh");
+
+    cpuid.set_memory_encryption_info(None).expect("can set leaf 8000_001Fh");
+
+    let mut leaf = ExtendedFeatureIdentification2::empty();
+    leaf.set_no_nested_data_bp(true);
+    leaf.set_lfence_always_serializing(true);
+    leaf.set_null_select_clears_base(true);
+    cpuid
+        .set_extended_feature_identification_2(Some(leaf))
+        .expect("can set leaf 8000_0021h");
+
+    let mut dump = cpuid.into_source();
+
+    // There are a few leaves that are not yet defined in `raw-cpuid` but we
+    // commit to being zero. In practice, *omitted* leaves with an explicit
+    // CPUID specification will be zero, but setting them to zero here avoids
+    // all doubt.
+
+    // First, instruction-based sampling (IBS) is hidden from guests for now
+    // (note `set_ibs(false)` above)
+    dump.set_leaf(0x8000_001B, Some(CpuIdResult::empty()));
+    // Lightweight profiling (LWP) is not supported by Milan, and not advertised
+    // to guests.  (note `set_lwp(false)` above)
+    dump.set_leaf(0x8000_001C, Some(CpuIdResult::empty()));
+    // SEV is not supported in guests (note `set_sev(false)` above)
+    dump.set_leaf(0x8000_001F, Some(CpuIdResult::empty()));
+
+    dump
+}
+
+pub fn milan_rfd314() -> CpuIdDump {
+    // This is the Milan we'd "want" to expose, absent any other constraints.
+    let baseline = milan_ideal();
+
+    let mut cpuid = CpuId::with_cpuid_reader(baseline);
+
+    let mut leaf = cpuid
+        .get_extended_feature_info()
+        .expect("baseline Milan defines leaf 1");
+
+    // RFD 314 describes the circumstances around RDSEED, but it is not
+    // currently available.
+    leaf.set_rdseed(false);
+
+    cpuid.set_extended_feature_info(Some(leaf)).expect("can set leaf 7h");
+
+    let mut leaf = cpuid
+        .get_extended_processor_and_feature_identifiers()
+        .expect("baseline Milan defines leaf 8000_0001");
+    // RFD 314 describes these leaf 8000_0001 wrinkles.
+    //
+    // Extended APIC space support was originally provided to guests because the
+    // host supports it and it was passed through. The extended space is not
+    // supported in Bhyve, but we leave it set here to not change it from under
+    // guests.
+    //
+    // Bhyve now supports all six performance counters, so we could set the perf
+    // counter extension bit here, but again it is left as-is to not change
+    // CPUID from under a guest.
+    //
+    // RDTSCP requires some Bhyve and Propolis work to support, so it is masked
+    // off for now.
```

Comment:
extended APIC support? that's a quirk. performance counter extension? quirk. RDTSCP? believe it or not, straight to quirks.